Resources/views/frontend/_public/src/js/jquery.product-click-tracking.js

```diff
@@ -0,0 +1,46 @@
+$.plugin('wbmProductClickTracking', {
+    init: function () {
+        var me = this,
+            i,
+            impressions = []
+
+        if (window.dataLayer) {
+            for (i = 0; i < window.dataLayer.length; i++) {
+                var layer = window.dataLayer[i];
+
+                if (layer.ecommerce && layer.ecommerce.impressions) {
+                    impressions = layer.ecommerce.impressions;
+                }
+            }
+        }
+
+        me._on(me.$el, 'click', $.proxy(me.onProductClicked, me, impressions));
+    },
+    /**
+        * @param {array} impressions
+        * @param {jQuery.Event} event
+    */
+    onProductClicked: function (impressions, event) {
+        let ordernumber = $(event.target).closest('.product--box').data('ordernumber');
+        let product = impressions.find(x => x.id === ordernumber);
+
+        window.dataLayer.push({
+            'event': 'productClick',
+            'ecommerce': {
+                'click': {
+                    'actionField': {'list': product.list},
+                    'products': [{
+                        'name': product.name,
+                        'id': product.id,
+                        'price': product.price,
+                        'brand': product.brand,
+                        'category': product.category,
+                        'position': product.position
+                    }]
```

Comment:
This is not a good solution since any of these used properties might have been removed through the backend user interface. Also any additional properties to the list impressions would be lost, rendering the whole point of having the template based editor useless.
First of all check whether `product` can  be found through the `id` which might not even exist or could've been altered by modifiers:

```
let product = impressions.find(x => x.id === ordernumber);

if (product === undefined) {
    return;
}
```

use the unaltered `product` that may include more or less properties in the array of `products`:
```
'products': [ product ]
```

Also this implementation will not cover the click tracking of products lazy loaded by scrolling past the corresponding threshold a.k.a infinite scrolling. 
There's an event that is being published once a new page has been fetched called `plugin/swInfiniteScrolling/onFetchNewPageFinished`. You may subscribe to that event and re-init the plugin. Also to allow click tracking of all currently visible products, you would have to merge all available instances of `layer.ecommerce.impressions` into one array before passing it to the click listener, as each new fetched page would represent a new array entry. Arguably though it would be better to store impressions within a property of the plugin and use that within the click callback:
```
this.impressions = ...
...
let product = this.impressions.find(...)
```

Reply:
Thanks for your support. I have taken note of your points:
- check if the product exists
- carry of additional properties
- infinite scrolling
I will solve these points before we merge this feature.